atlp-rwanda/eagles-ec-be
# #23187419173 - fx:fixed Two Factor Authentication for Sellers

#### What does this PR do?

This PR sets up Two Factor Authentication for Sellers

#### Description of Task to be completed?

-GIVEN there is a request to log in a user to the website
WHEN 2FA is enabled on the particular user
THEN Further request for a 2FA token should be requested and once the token is received and validated, the user is allowed to log in.


#### How should this be manually tested?

1.  checkout this branch and run app 
2. regiter new user and modify user role to seller
3. then login with email and password
4. there u will see magic happen u will get message that verification token has been sent to ur email with verification link
5. go there open email and copy token and send them in req.body to specified endpoint in body `/api/v1/users/2fa-verify` 


####  pivotal tracker id ----> #23187419173
#### Screenshots 


<img width="548" alt="Screenshot 2024-05-03 at 19 34 57" src="https://github.com/atlp-rwanda/eagles-ec-be/assets/105858896/5d6d40f1-0c0b-4e6a-bcbe-5932f6bc6df6">
<img width="541" alt="Screenshot 2024-05-03 at 20 08 45" src="https://github.com/atlp-rwanda/eagles-ec-be/assets/105858896/9278b58c-7ff5-48a4-9adb-a82a2ecee534">
<img width="769" alt="Screenshot 2024-05-03 at 20 09 04" src="https://github.com/atlp-rwanda/eagles-ec-be/assets/105858896/9e3a369f-eb00-40a2-924a-84d6106d5cf9">
